spark_log_parser/loaders/json.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import orjson
+from typing import Generic, TypeVar
+
+from aiodataloader import DataLoader
+
+from spark_log_parser.loaders.https import HTTPFileLinesDataLoader, HTTPFileBlobDataLoader
+from spark_log_parser.loaders.local_file import LocalFileBlobDataLoader, LocalFileLinesDataLoader
+from spark_log_parser.loaders.s3 import S3FileBlobDataLoader, S3FileLinesDataLoader
+
+
+RawJSONBlobDataLoader = TypeVar("BlobDataLoader", LocalFileBlobDataLoader, S3FileBlobDataLoader, HTTPFileBlobDataLoader)
+
+
+class JSONBlobDataLoader(DataLoader, Generic[RawJSONBlobDataLoader]):
+    blob_data_loader: RawJSONBlobDataLoader
+
+    def __init__(self, blob_data_loader: RawJSONBlobDataLoader, **kwargs):
+        super().__init__(**kwargs)
+        self.blob_data_loader = blob_data_loader
+
+    async def batch_load_fn(self, keys):
+        raw_datas = await self.blob_data_loader.load_many(keys)
+        # We expect each "blob" here to be well-formed JSON, so parse each of them thusly
+        return [orjson.loads(next(raw_data)) for raw_data in raw_datas]
+
+
+RawJSONLinesDataLoader = TypeVar("LinesDataLoader", LocalFileLinesDataLoader, S3FileLinesDataLoader,
+                                 HTTPFileLinesDataLoader)
+
+
+class JSONLinesDataLoader(DataLoader, Generic[RawJSONLinesDataLoader]):
+    cache = False
+    lines_data_loader: RawJSONLinesDataLoader
+
+    def yield_json_lines(self, filename, lines):
+        num_bad_lines_seen = 0
+        for line in lines:
+            try:
+                yield orjson.loads(line)
+            # Note - this is largely here because we may get arbitrary file types in archives where we are
+            #  expecting eventlog files (which are JSON Lines). If we try to load those "lines" as
+            #  JSON objects, they will fail, and so this allows us to just skip those "bad" lines and
+            #  continue processing other files in the archive. However, this `except` could maybe be
+            #  more robust so that we aren't ignoring real issues! If we do encounter some bad lines,
+            #  we will at the very least log how many we hit
+            except orjson.JSONDecodeError:
+                num_bad_lines_seen += 1
+                # We have to continue here because these will be the lines for the top-level file. If this is an
+                #  archive, then it is possible for some lines to not be parse-able as JSON, but for others to be
+                continue
+
+        if num_bad_lines_seen > 0:
+            self.logger.info(f"Failed to parse {num_bad_lines_seen} JSON lines in top-level file: {filename}")
```

Comment:
The `databricks-rollover-messy` test case has an example of just a regular `.json` file in it, which is the databricks pricing plan information, which can't be parsed as JSON lines (which is to say - we can't assume that each line of the file is a well-formed JSON object). I do have a change staged locally that handles this a little differently for handing "parsed" logs vs. "un-parsed" logs, though

Reply:
Ah yeah. Wouldn't we still want to bail on the file though? If it's an event log file and we can't parse a record we should throw the whole thing out - the integrity of the log and anything based on it is gone. If it's a pricing file then we shouldn't continue to try to parse lines from it...

Reply:
yeah - I address this a little bit differently in that diff I have locally. If we want to do something about it in this PR, I can pull that in - I just didn't necessarily want to make this diff even larger than it already is, haha

Reply:
In that diff, instead of just yielding raw lines, we yield a tuple of `(filename, file_chunks)` so that we can do what you are describing, i.e. just skip to the next file if we'd like to. Right now it's hard to tell because at this level we don't know where one file in an archive ends and another one begins.